### PR TITLE
DRAFT: Add Context Explorer View

### PR DESCRIPTION
**Pull request by dayo09**
This draft implements context explorer view.

For #494
From #480

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -53,6 +54,8 @@ export function activate(context: vscode.ExtensionContext) {
 
   setGlobalContext();
 
+  new ContextExplorer(context);
```
Comment (jyoungyun) [inline src/extension.ts:57]: If we determine the view name, it would be better to change this class and file name.

File: src/contextExplorer.ts [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+import {config} from 'process';
+import {stringify} from 'querystring';
+import * as vscode from 'vscode';
+import { Model } from './Circlereader/circle-analysis/circle/model';
+
+enum NodeType{
+  directory,
+  model,
+  config,
+  configWrapper
```
Comment (jyoungyun) [inline src/contextExplorer.ts:28]: (Q) Do we need config wrapper directory? How about showing `cfg` file directly without a config wrapper director under the model file?

File: src/contextExplorer.ts [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+}
+
+interface Node {
+  type: NodeType, name: string, childs: Node[]
+}
+
+interface DirNode extends Node{
```
Comment (jyoungyun) [inline src/contextExplorer.ts:35]:
```suggestion
interface DirNode extends Node {
```

File: src/contextExplorer.ts [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+  uri: vscode.Uri
+}
+
+interface ModelNode extends Node {
+  uri: vscode.Uri
+}
+
+interface ConfigNode extends Node {
+  uri: vscode.Uri
+  pairModel: Node
```
Comment (jyoungyun) [inline src/contextExplorer.ts:45]:
```suggestion
  pairModel: ModelNode
```

File: src/contextExplorer.ts [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+export class ContextNode extends vscode.TreeItem {
+  constructor(
+      public readonly label: string,
+      public readonly collapsibleState: vscode.TreeItemCollapsibleState,
+      public readonly node: Node,
+  ) {
+    super(label, collapsibleState);
+
+    this.tooltip = `${this.label}`;
```
Comment (jyoungyun) [inline src/contextExplorer.ts:59]: How about showing full file path like file explorer view?

File: src/contextExplorer.ts [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+      if (fstat.isDirectory()) {
+        const dirNode = {type: NodeType.directory, name: fn, dir: true, childs: [], uri: vscode.Uri.file(fpath)};
+
+        this.searchConfigs(dirNode, dirpath);
+        if (dirNode.childs.length > 0) {
+          node.childs.push(dirNode);
+        }
+      }
+      else if ((fstat.isFile() 
+        && (fn.endsWith('.tflite')) || fn.endsWith('.onnx'))){
```
Comment (jyoungyun) [inline src/contextExplorer.ts:149]: How about supporting the tensorflow model ?

File: src/contextExplorer.ts [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import {config} from 'process';
+import {stringify} from 'querystring';
+import * as vscode from 'vscode';
+import { Model } from './Circlereader/circle-analysis/circle/model';
```
Comment (jyoungyun) [inline src/contextExplorer.ts:22]:
```suggestion
import * as fs from 'fs';
import * as path from 'path';
import * as vscode from 'vscode';
```

File: src/contextExplorer.ts (hunk 2 above)
Comment (dayo09) [reply src/contextExplorer.ts]: Sure. @hyunsik-yoon game me the same opinion, I have removed the wrapper node.

File: src/contextExplorer.ts (hunk 5 above)
Comment (dayo09) [reply src/contextExplorer.ts]: Looks like a good idea.

**Comment by dayo09** [comment]
https://github.com/Samsung/ONE-vscode/pull/505 is merged

**Pull request closed without merging**